index.js

```diff
@@ -40,6 +41,25 @@ class ScmBase {
         this.config = config;
     }
 
+    /**
+     * Set token correctly if is read-only SCM
+     * @param  {Object} config
+     * @param  {String} [token]   SCM token
+     * @return {Object}           Config with proper token
+     */
+    getConfig(config) {
+        const newConfig = config;
+        const readOnlyToken = Hoek.reach(this.config, 'readOnly.accessToken');
```

Comment:
config already contains a `username` & `commentUserToken`. Is this newly added readOnly access token end up being same as `commentUserToken` ? Should we have just one token ?

Reply:
So here are my thoughts about that:
- ideally it'd be nice to have just one token but the `commentUserToken` is supposed to be strictly used for PR comments
- if we want to overload it, I can change this; might cause breaking changes?
- also having a separate username is nice so that it's clear that everything in that section is for readOnly purpose, like this:
```
scms:
  gitlab:
    username: username
    commentUserToken: commentToken
    readOnly:
      enabled: true
      username: readOnlyUsername
      accessToken: readOnlyToken
```

If you want just one of everything, could have something like this:
```
scms:
  gitlab:
    username: username
    commentUserToken: token
    readOnlyEnabled: true
```